Moved Expenses from ExpenseListServlet to ExpenseService

CameronCobb opened the pull request.

CameronCobb committed:
Moved Expenses from ExpenseListServlet to ExpenseService

andrewosipenko commented on src/main/java/com/es/jointexpensetracker/service/ExpenseService.java:
```diff
@@ -0,0 +1,68 @@
+        return expenses;
+    }
+
+    public void setExpenses(ArrayList<Expense> expenses) {
```
I wonder if we need this method. All modifications can be done via addExpense()/removeExpense()

CameronCobb committed:
Removed method addExpenses() from  ExpenseService.

CameronCobb committed:
Changed href in page.tag for correct redirection.
Added expense.jsp for expense details.
Added page404.jsp for incorrect url's.
Implemented method doGet() in ExpenseServlet

CameronCobb committed:
Task 2.4
Removed modifier private in Expense
Implemented doPost and some extra methods in ExpenseServlet
Added <form> in expense.jsp

CameronCobb committed:
Task 2.5
Added FlashScopeFilter to handle messages
Have refactored code a little bit

CameronCobb committed:
Task 2.6
Implemented DeleteExpenseServlet extends ExpenseServlet
Did some code refactoring

CameronCobb committed:
Update ExpenseServlet.java

CameronCobb committed:
Update expense.jsp

CameronCobb committed:
Task 2.7
Implemented AddExpenseServlet extends ExpenseServlet
Moved code form doPost of ExpenseServlet to separated methods
Added default constructor to Expense, changed ID system

CameronCobb committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/webapp/WEB-INF/pages/expense.jsp

CameronCobb committed:
Corrected expense.jsp

CameronCobb committed:
Code refactoring
Moved all extra parse methods from ExpenseServlet to ParseExpenseService
Implemented MessageService

CameronCobb committed:
Task 2.7, 2.8
Implemented StatisticsServlet
Implemented Debt class

CameronCobb committed:
Removed if statement

CameronCobb committed:
Removed id from Expense() arguments

CameronCobb committed:
Removed unnecessary imports

CameronCobb committed:
Moved brackets to new line

CameronCobb committed:
Moved brackets to new line

CameronCobb committed:
Moved brackets to new line

CameronCobb committed:
Deleted empty rows

CameronCobb committed:
Deleted empty rows, moved brackets to new line, added spaces

CameronCobb committed:
Moved brackets to new line

CameronCobb committed:
Moved brackets to new line

CameronCobb committed:
Moved brackets to new line

CameronCobb committed:
Implemented Payment class

CameronCobb committed:
Implemented DebtsService

CameronCobb committed:
Moved debts calculation to DebtsService

CameronCobb committed:
Add compiler property

CameronCobb committed:
Tasks 2.10, 2.11
Implemented CorrectURIFilter for checking right of URI's
Replaced List with Map
Implemented ExpenseGroupUUIDService for handling UUID
Implemented URLpatternHandler for handling url pattern's
Implemented ExpenseGroupServlet for handling request's with UUID
Implemented HomePageServlet for index.jsp

CameronCobb committed:
Removed unnecessary comment

CameronCobb committed:
Deleted previous ExpenseListServlet

CameronCobb committed:
Moved all code in ExpenseGroupServlet in try-block to handle extra exceptions

CameronCobb committed:
Removed println()
